diff --git a/src/weewxd.py b/src/weewxd.py
index 185625a6..8b1c8817 100644
--- a/src/weewxd.py
+++ b/src/weewxd.py
@@ -187,7 +187,7 @@ def main():
             log.info("Terminating weewx version %s", weewx.__version__)
             weeutil.logger.log_traceback(log.debug, "    ****  ")
             signal.signal(signal.SIGTERM, signal.SIG_DFL)
-            os.kill(0, signal.SIGTERM)
+            sys.exit(0)
 
         # Catch any keyboard interrupts and log them
         except KeyboardInterrupt:

